pkg/piecestore/rpc/client/pieceranger_test.go

```diff
@@ -108,24 +112,27 @@ func TestPieceRangerSize(t *testing.T) {
 		errTag := fmt.Sprintf("Test case #%d", i)
 
 		route := pb.NewMockPieceStoreRoutesClient(ctrl)
+		pid := NewPieceID()
+
+		stream := pb.NewMockPieceStoreRoutes_RetrieveClient(ctrl)
+
 		if tt.offset >= 0 && tt.length > 0 && tt.offset+tt.length <= tt.size {
-			stream := pb.NewMockPieceStoreRoutes_RetrieveClient(ctrl)
 			gomock.InOrder(
-				route.EXPECT().Retrieve(
-					gomock.Any(), gomock.Any(), gomock.Any(),
-				).Return(stream, nil),
+				stream.EXPECT().Send(&pb.PieceRetrieval{PieceData: &pb.PieceRetrieval_PieceData{Id: pid.String(), Size: tt.length, Offset: tt.offset}}).Return(nil),
+				stream.EXPECT().Send(&pb.PieceRetrieval{Bandwidthallocation: &pb.BandwidthAllocation{Data: &pb.BandwidthAllocation_Data{Payer: "payer-id", Renter: "renter-id", Size: 32 * 1024, Total: 32 * 1024}}}).Return(nil),
 				stream.EXPECT().Recv().Return(
 					&pb.PieceRetrievalStream{
-						Size:    tt.size,
+						Size:    tt.length,
 						Content: []byte(tt.data)[tt.offset : tt.offset+tt.length],
 					}, nil),
+				stream.EXPECT().Send(&pb.PieceRetrieval{Bandwidthallocation: &pb.BandwidthAllocation{Data: &pb.BandwidthAllocation_Data{Payer: "payer-id", Renter: "renter-id", Size: 32 * 1024, Total: 32 * 1024 * 2}}}).Return(nil),
 				stream.EXPECT().Recv().Return(&pb.PieceRetrievalStream{}, io.EOF),
 			)
 		}
 
 		ctx := context.Background()
```

Comment:
set var ( ctx := context.Backgroud() ) that was you don't have to call it every time. Set it as a global.  That's what I did for my stuff.